snapshot/src/main/java/io/camunda/zeebe/snapshots/impl/FileBasedSnapshotStore.java

```diff
@@ -46,30 +46,32 @@
 public final class FileBasedSnapshotStore extends Actor
     implements ConstructableSnapshotStore, ReceivableSnapshotStore {
 
+  static final int VERSION = 1;
+
+  // When sorted with other files in the snapshot, the metadata file must be ordered at the end.
+  // This is required for backward compatibility of checksum calculation. Otherwise, the older
+  // versions, which are not aware of the metadata will calculate the checksum using a different
+  // order of files. We can change the name in later versions, because the new checksum calculation
+  // already order the metadata file explicitly instead of using the implicit sort order.
+  static final String METADATA_FILE_NAME = "zeebe.metadata";
```

Comment:
:question: If I understand correctly, the ordering requirement is fulfilled because the name is "zeebe.metadata" and we know that this is lexicographically greater than all other snapshot files, right? Could we add a comment that clarifies that a little bit?

Reply:
Yes. I have updated the comment.